### PR TITLE
Add seen/unseen state in Threads list UI

### PR DESCRIPTION
**Pull request by KevinBoulongne**
Depends on #110

### CODE REVIEW

File: app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadListAdapter.kt [hunk 1]
```diff
@@ -97,16 +102,40 @@ class ThreadListAdapter : RecyclerView.Adapter<ViewHolder>() { // TODO: Use Load
 
     private fun displayThread(binding: ItemThreadBinding, position: Int) = with(binding) {
         val thread = itemsList[position] as Thread
+        val context = itemThread.context
+
         expeditor.text = thread.from[0].name.ifEmpty { thread.from[0].email }
```
Comment (sirambd) [inline app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadListAdapter.kt:107]: use `kotlin.first`

File: app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadListAdapter.kt [hunk 2]
```diff
@@ -97,16 +102,40 @@ class ThreadListAdapter : RecyclerView.Adapter<ViewHolder>() { // TODO: Use Load
...
+        } else {
+            R.color.primaryText to R.font.suisseintl_regular
+        }
+        val textColor = ContextCompat.getColor(context, textColorRes)
+        val typeFace = ResourcesCompat.getFont(context, typeFaceRes)
+
+        fun TextView.applyState(@IdRes textColor: Int, typeFace: Typeface?) {
+            setTextColor(textColor)
+            typeface = typeFace
+        }
```
Comment (sirambd) [inline app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadListAdapter.kt:133]: Check if exists please

File: app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadListFragment.kt [hunk 3]
```diff
@@ -171,9 +173,11 @@ class ThreadListFragment : Fragment() {
             displayThreadList()
         }
 
-        with(threadListAdapter) {
-            val newList = formatList(threads, requireContext())
-            notifyAdapter(newList)
+        context?.let {
+            with(threadListAdapter) {
+                val newList = formatList(threads, it)
+                notifyAdapter(newList)
```
Comment (sirambd) [inline app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadListFragment.kt:179]: merge this

**Review by sirambd — CHANGES_REQUESTED** [review]
Waiting for request changes

**Review by sirambd — APPROVED** [review]

File: app/src/main/res/layout/item_thread.xml [hunk 4]
```diff
@@ -117,7 +115,8 @@
                 android:layout_height="match_parent"
                 android:ellipsize="end"
                 android:maxLines="1"
-                android:text="TODO short body when api returns it"
+                android:text="Display the short body when the API returns it"
```
Comment (JorisBodin) [inline app/src/main/res/layout/item_thread.xml:118]: Why isn't it in "tools"?

File: app/src/main/res/layout/item_thread.xml [hunk 5]
```diff
@@ -55,9 +54,10 @@
                 android:ellipsize="end"
                 android:maxWidth="235dp"
                 android:maxLines="1"
-                android:text="expeditor name looooooooooooooooooooooooooooooong"
+                android:textSize="18sp"
                 app:layout_constraintStart_toStartOf="parent"
-                app:layout_constraintTop_toTopOf="parent" />
+                app:layout_constraintTop_toTopOf="parent"
+                tools:text="Expeditor name" />
```
Comment (JorisBodin) [inline app/src/main/res/layout/item_thread.xml:60]: Use : tools:text="@tools:sample/full_names"

**Review by JorisBodin — APPROVED** [review]

**Pull request merged**